### DISCUSSION TITLE
During an Upsert (insert ... on conflict) which RLS policies apply?

### DISCUSSION THREAD

**Discussion by olirice** [supabase/supabase#28122] [Questions] [answered]
When performing an upsert (insert ... on conflict) do the RLS (row level security) policies for `insert` and `update` both need to pass?

Do different policies apply if there is a conflict vs no conflict?

**Comment by olirice** [comment, marked as answer]
If an `on conflict` clause is specified in an insert statement

e.g.
```sql
insert into blog_post(id, title, status)
  values ('f173924c-9be2-447a-b97b-b37a6ed16ca7', 'PUBLISHED')
  on conflict (id) -- THIS CLAUSE
  do update
    set
        status = excluded.status,
        title = excluded.title;
```

all applicable `select`, `insert` and `update` policies must return `true` for the upsert to succeed regardless of if there is a conflict during the upsert.

**Reply by GaryAustin1** [reply]
It will also need to meet the `select` policy I'm pretty sure.

**Reply by olirice** [reply]
good point! I was assuming that'd be the most permissive. updated the comment
